index.js

```diff
@@ -1,9 +1,87 @@
-var crypto = require('crypto')
+var crypto = require('crypto');
+var fork = require('child_process').fork;
+var path = require('path');
+var compat = require('./browser');
+var hashes = {
+  sha1: 0,
+  0: 'sha1',
+  sha224: 1,
+  1: 'sha224',
+  sha256: 2,
+  2: 'sha256',
+  sha384: 3,
+  3: 'sha384',
+  sha512: 4,
+  4: 'sha512'
+};
+function asyncpbkdf2(password, salt, iterations, keylen, digest, callback) {
+  if (typeof iterations !== 'number') {
+    throw new TypeError('Iterations not a number')
+  }
+  if (iterations < 0) {
+    throw new TypeError('Bad iterations')
+  }
 
-var exportFn = require('./pbkdf2')
-var exported = exportFn(crypto)
+  if (typeof keylen !== 'number') {
+    throw new TypeError('Key length not a number')
+  }
 
-module.exports = {
-  pbkdf2: exported.pbkdf2,
-  pbkdf2Sync: exported.pbkdf2Sync
+  if (keylen < 0) {
+    throw new TypeError('Bad key length')
+  }
+  var msg = {
+    password: password.toString(),
+    salt: salt.toString(),
+    iterations: iterations,
+    keylen: keylen,
+    digest: digest
+  };
+
+  var child = fork(path.resolve(__dirname, 'async-shim.js'));
```

Comment:
no node does it nativly on a thread pool but this the only way to make an async process that doesn't block I could think of.